Remove FilterUtil to avoid type cast problems

HolgerKache opened the pull request:
BugzId: 55905

```
modified:   src/main/scala/com/cloudant/spark/common/FilterUil.scala
```

HolgerKache committed:
Remove call to FilterUtil

Clean up FilerUtil but leave in project for futurereference

BugzId: 55905

	modified:   src/main/scala/com/cloudant/spark/common/FilterUil.scala
	modified:   src/main/scala/com/cloudant/spark/common/JsonStoreDataAccess.scala

	modified:   cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala
	modified:   cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreDataAccess.scala

HolgerKache committed:
Cast JSON number to Scala long instead of int

BugzId: 55905

	modified:   src/main/scala/com/cloudant/spark/common/FilterUil.scala

yanglei99 commented:
The solution should take care of both int and long case by adding new comparing types in the case (.., v: Int) or case (.., v:Long), instead of replacing int with long.

HolgerKache committed:
Support both, integer and long in separate cases

	modified:   src/main/scala/com/cloudant/spark/common/FilterUil.scala

mayya-sharipova commented on cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala:
```diff
@@ -134,7 +134,9 @@ class FilterUtil(filters: Map[String, Array[Filter]]){
         {
-          if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.intValue(), filters)
+          if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.longValue(), filters)
+          else if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.intValue(), filters)
```
The control flow will never go to line 138, because if field.isInstanceOf[JsNumber] it will always go to 137.

mayya-sharipova replied:
I would have just one clause for JsNumber:
if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.doubleValue(), filters)

mayya-sharipova commented on cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala:
```diff
@@ -134,7 +134,9 @@ class FilterUtil(filters: Map[String, Array[Filter]]){
+          if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.longValue(), filters)
+          else if (field.isInstanceOf[JsNumber])  satisfiesAll(field.as[JsNumber].value.intValue(), filters)
+          else if (field.isInstanceOf[JsBoolean])  satisfiesAll(field.as[JsBoolean].value, filters)
           else if (field.isInstanceOf[JsBoolean])  satisfiesAll(field.as[JsBoolean].value, filters)
```
The same is true about line 140, the control flow will never go to this line, so this line ins redundant. Not sure, why we need at JsBoolean clauses any way

HolgerKache committed:
Add support for all numeric types available in Scala

	modified:   cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala

HolgerKache committed:
Improve formatting

	modified:   cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala

HolgerKache committed:
Fix identation

	modified:   cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala

HolgerKache committed:
Remove the FilterUtil functions and let Spark apply any filters

	modified:   cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/FilterUil.scala

HolgerKache committed:
Merge branch '55905_long_type_cast' of https://github.com/cloudant-labs/spark-cloudant into 55905_long_type_cast

HolgerKache committed:
More formatting changes

	modified:   src/main/scala/com/cloudant/spark/common/FilterUil.scala

mayya-sharipova commented:
+1 with Nov. 17 changes, looks great to me
